R/draw_binary_icc.R

```diff
@@ -0,0 +1,97 @@
+#' Draw binary data with fixed intra-cluster correlation.
+#'
+#' Data is generated to ensure inter-cluster correlation 0, intra-cluster
+#' correlation in expectation \eqn{\rho}. Algorithm taken from Hossein,
+#' Akhtar. "ICCbin: An R Package Facilitating Clustered Binary Data
+#' Generation, and Estimation of Intracluster Correlation Coefficient (ICC)
+#' for Binary Data".
+#'
+#' @param x A number or vector of numbers, one probability per cluster.
+#' @param N (Optional) A number indicating the number of observations to be
+#' generated. Must be equal to length(clusters) if provided.
+#' @param clusters A vector of factors or items that can be coerced to
+#' clusters; the length will determine the length of the generated data.
+#' @param rho A number indicating the desired RCC.
+#' @return A vector of binary numbers corresponding to the observations from
+#' the supplied cluster IDs.
+#' @examples
+#' clusters = rep(1:5, 10)
+#' draw_binary_icc(clusters = clusters)
+#' draw_binary_icc(x = 0.5, clusters = clusters, rho = 0.5)
+#'
+#' @importFrom stats rbinom
+#'
+#' @export
+draw_binary_icc = function(x = 0.5, N = NULL, clusters, rho = 0.5) {
```

Comment:
Mild preference for 0 or 1 to be the default rho here and for normal.

Reply:
Agreed.